### ISSUE TITLE
omni erroneously reports "The AM speaks the wrong API version, not 2"

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-tools#275]
omni reports "The AM speaks the wrong API version, not 2" when:
 * The user is not trusted at an aggregate
 * The aggregate is not running

 _Imported from trac ticket [#275](http://trac.gpolab.bbn.com/gcf/ticket/275),  created by tmitchel on 05-21-2013 at 15:59, last modified: 08-30-2013 at 14:40_

**Comment by ahelsing** [comment]
checkValidVersion returns a None for the client in various cases. All these are treated by _api_call as a BadClientException - the only way it can currently distinguish is via the message.

And createsliver treats a BadClientException as an error with the message you are seeing.

Sigh.


_Trac comment by ahelsing on 05-21-2013 at 16:04_

**Comment by ahelsing** [comment]
Fixed.

Omni looks for particular error messages from dossl (like Connection Refused). For those cases, it stops trying to contact that AM, and highlights the real error message.

In the process, I cleaned up some log messages, made createsliver get its client more like other calls, and reduced calls to _listaggregates.

To test, run getversion, listresources, createsliver, and sliverstatus with
 - 1 good client
 - 1 bad client (like a made up URL, or a use a real AM but your GCF user cert)
 - 1 bad client followed by 1 good client
 - 1 good client followed by 1 bad client
 - 3 clients: 1 good, 1 bad, 1 good

When running with 1 bad client, Omni should exit abruptly. For the other cases, Omni should do the command at the good clients, and tell you it removed the bad client.


_Trac comment by ahelsing on 08-30-2013 at 14:40_
